exchanges/cryptodotcom/cryptodotcom_wrapper.go

```diff
@@ -0,0 +1,966 @@
+			Subscriber:            cr.Subscribe,
+			Unsubscriber:          cr.Unsubscribe,
+			GenerateSubscriptions: cr.GenerateDefaultSubscriptions,
+			Features:              &cr.Features.Supports.WebsocketCapabilities,
+		})
+	if err != nil {
+		return err
+	}
+	err = cr.Websocket.SetupNewConnection(stream.ConnectionSetup{
+		URL:                  cryptodotcomWebsocketMarketAPI,
+		ResponseCheckTimeout: exch.WebsocketResponseCheckTimeout,
+		ResponseMaxLimit:     exch.WebsocketResponseMaxLimit,
+	})
+	if err != nil {
+		return err
+	}
+	return cr.Websocket.SetupNewConnection(stream.ConnectionSetup{
+		URL:                  cryptodotcomWebsocketUserAPI,
+		ResponseCheckTimeout: exch.WebsocketResponseCheckTimeout,
+		ResponseMaxLimit:     exch.WebsocketResponseMaxLimit,
+		Authenticated:        true,
+	})
+}
+
+// Start starts the Cryptodotcom go routine
+func (cr *Cryptodotcom) Start(wg *sync.WaitGroup) error {
+	if wg == nil {
+		return fmt.Errorf("%T %w", wg, common.ErrNilPointer)
+	}
+	wg.Add(1)
+	go func() {
+		cr.Run()
+		wg.Done()
+	}()
+	return nil
+}
+
+// Run implements the Cryptodotcom wrapper
+func (cr *Cryptodotcom) Run() {
+	if cr.Verbose {
+		log.Debugf(log.ExchangeSys,
+			"%s Websocket: %s.",
+			cr.Name,
+			common.IsEnabled(cr.Websocket.IsEnabled()))
+		cr.PrintEnabledPairs()
+	}
+
+	if !cr.GetEnabledFeatures().AutoPairUpdates {
+		return
+	}
+
+	err := cr.UpdateTradablePairs(context.TODO(), false)
+	if err != nil {
+		log.Errorf(log.ExchangeSys,
+			"%s failed to update tradable pairs. Err: %s",
+			cr.Name,
+			err)
+	}
+}
+
+// FetchTradablePairs returns a list of the exchanges tradable pairs
+func (cr *Cryptodotcom) FetchTradablePairs(ctx context.Context, a asset.Item) (currency.Pairs, error) {
+	if !cr.SupportsAsset(a) {
+		return nil, fmt.Errorf("asset type of %s is not supported by %s", a, cr.Name)
+	}
+	instruments, err := cr.GetInstruments(ctx)
+	if err != nil {
+		return nil, err
+	}
+	pairs := make(currency.Pairs, len(instruments))
+	for x := range instruments {
+		cp, err := currency.NewPairFromString(instruments[x].InstrumentName)
+		if err != nil {
+			return nil, err
+		}
+		pairs[x] = cp
+	}
+	return pairs, nil
+}
+
+// UpdateTradablePairs updates the exchanges available pairs and stores
+// them in the exchanges config
+func (cr *Cryptodotcom) UpdateTradablePairs(ctx context.Context, forceUpdate bool) error {
+	pairs, err := cr.FetchTradablePairs(ctx, asset.Spot)
+	if err != nil {
+		return err
+	}
+	return cr.UpdatePairs(pairs, asset.Spot, false, forceUpdate)
+}
+
+// UpdateTicker updates and returns the ticker for a currency pair
+func (cr *Cryptodotcom) UpdateTicker(ctx context.Context, p currency.Pair, assetType asset.Item) (*ticker.Price, error) {
+	tickerPrice := new(ticker.Price)
+	tick, err := cr.GetTicker(ctx, p.String())
+	if err != nil {
+		return tickerPrice, err
+	}
+	if len(tick.Data) != 1 {
+		return tickerPrice, errInvalidResponseFromServer
+	}
+	tickerPrice = &ticker.Price{
```

Comment:
`new(ticker.Price)` allocates and is a non-nil value. We can just remove that above and do a short hand `:=` operator.